mobile/src/components/billing/components/payment-screen/payment-screen.tsx

```diff
@@ -0,0 +1,106 @@
+import { CardField, useStripe } from '@stripe/stripe-react-native';
+import React, { FC } from 'react';
+
+import { AppColor } from '~/common/enums/enums';
+import { BillingReplenishToken } from '~/common/types/types';
+import { ReplenishAmount } from '~/components/billing/types/types';
+import { Button, ScrollView, Text, View } from '~/components/common/common';
+import { useState } from '~/hooks/hooks';
+
+import { ReplenishCard } from './components/components';
+import { styles } from './styles';
+
+type Props = {
+  replenishKey: string;
+  replenishingPricesDtos: readonly ReplenishAmount[];
+  moneyBalance: number;
+  onReplenish: (
+    amountOfMoneyToReplenish: number,
+    token: BillingReplenishToken,
+  ) => void;
+  onWithdraw: () => void;
+};
+
+const PaymentScreen: FC<Props> = ({
+  replenishingPricesDtos,
+  moneyBalance,
+  onReplenish,
+  onWithdraw,
+}) => {
+  const userMoneyBalance = `$${moneyBalance}`;
+  const { createToken } = useStripe();
+  const [selectedCardId, setSelectedCardId] = useState<number | null>(null);
+
+  const handleReplenishCardChoose = (cardId: number): void => {
+    const selectedCard = selectedCardId !== cardId ? cardId : null;
+
+    setSelectedCardId(selectedCard);
+  };
+
+  const handlePayPress = async (): Promise<void> => {
+    const { token: createdToken } = await createToken({
+      type: 'Card',
+      currency: 'USD',
+    });
+    const amountOfMoneyToReplenish = replenishingPricesDtos.find(
+      (dto) => dto.id === selectedCardId,
+    );
+
+    if (createdToken && amountOfMoneyToReplenish) {
+      onReplenish(amountOfMoneyToReplenish.value, createdToken);
+    }
+
+    return;
+  };
+
+  return (
+    <ScrollView>
+      <View style={styles.container}>
+        <Text style={styles.balanceAmount}>{userMoneyBalance}</Text>
+        <Text style={styles.balanceText}>Current balance</Text>
+        <View style={styles.buttonWrapper}>
+          <Button label="Withdraw" onPress={onWithdraw} />
+        </View>
+      </View>
+      <View style={styles.cardsContainer}>
+        {replenishingPricesDtos.map((dto) => {
+          return (
+            <ReplenishCard
+              key={dto.id}
+              id={dto.id}
+              selectedCardId={selectedCardId}
+              amount={dto.value}
+              onCardChoose={(): void => handleReplenishCardChoose(dto.id)}
+            />
+          );
+        })}
+      </View>
+      {selectedCardId && (
+        <View style={styles.container}>
+          <CardField
```

Comment:
Maybe it would be better to create a separate component, styles in another folder and default number '4242 4242 4242 4242' to constants